src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,180 @@
+        Cart cartNull = new Cart(null);
+        testParser.writeToFile(cartNull);
+
+        assertTrue(Files.deleteIfExists(fileToDeletePath));
+    }
+
+
+
+    @Test
+    @Tag("JsonParser")
+    @Tag("Write_to_file")
+    @Tag ("Smoke")
+    @DisplayName("File contains correct data from Object")
+    public void writeToFileContainsCorrectData() throws IOException {
+        testParser.writeToFile(testCart);
+
+        Reader reader = new FileReader("src/main/resources/CartName.json");
+        Cart createdCart = gson.fromJson(reader, Cart.class);
+        reader.close();
+
+        assertEquals(testCart.getCartName(), createdCart.getCartName(), "Wrong information was " +
+                "written in the file");
+    }
+
+
+
+    //Disabled case!!
+    @Test
+    @Tag("JsonParser")
+    @Disabled
+    @Tag("Write_to_file")
+    @Tag ("Critical_Path")
+    @DisplayName("Exception because file is read only")
+    public void writeToFileReadOnly() throws IOException {
+        newFile.createNewFile();
+        newFile.setReadOnly();
+        System.out.println(newFile.createNewFile());
+        assertThrows(IOException.class, () -> testParser.writeToFile(testCart));
+    }
+
+
+
+    // TODO: 7/23/2021
+    @Test
+    @Tag("JsonParser")
+    @Tag("Read_From_File")
+    @Tag ("Smoke")
+    @DisplayName("Positive test, reading data from file")
+    public void readFromFileSmoke() throws IOException {
+        File eugenCart = new File("src/main/resources/eugen-cart.json");
+
+        Reader reader = new FileReader("src/main/resources/eugen-cart.json");
+        Cart expectedCart = gson.fromJson(reader, Cart.class);
+        reader.close();
+        Cart actualCart = testParser.readFromFile(eugenCart);
+
+        assertAll("Values",
+                () -> assertEquals(expectedCart.getCartName(), actualCart.getCartName(), "incorrect cart " +
+                        "name value"),
+                () -> assertEquals(expectedCart.getTotalPrice(), actualCart.getTotalPrice(), "Incorrect total " +
+                        "Price" )
+        );
+    }
+
+
+
+    @ParameterizedTest
+    @Tag("JsonParser")
+    @Tag("Read_From_File")
+    @Tag ("Critical_Path")
+    @DisplayName("Parametrized Test: Reading data from unexciting files")
+    @ValueSource(strings = {
+            "empty",
+            "src\\main\\\\rsources\\\\cart.json",
+            "src/main/resources/CartName",
+            "/vCartName.json",
+            "src/main/resources/andrew-cartn.json",
+            "src/main/resources/CartNme",
+            "C:\\Users\\cloud_qa1\\.android",})
+
+    void testWithValueSource(String filePath) {
+        newFile = new File(filePath);
+        // System.out.println(newFile.getAbsolutePath());
+        Exception exception = assertThrows(NoSuchFileException.class, () ->
+                testParser.readFromFile(newFile));
+
+        assertTrue(exception.getMessage().contains("not found!"));
```

Comment:
Add message on failure as additional parameter

Reply:
Message was added into the Exception test
commit 0300288f1690ed0b5453336290d4447e045ab4b0